### DISCUSSION TITLE
Why does pkg give the error ERR_REQUIRE_ESM when I can get my code to work with node ./index.js ?

### DISCUSSION THREAD

**Discussion by MGMehdi** [yao-pkg/pkg#84] [Q&A]
Everything is in the title. my node version is 20.12.0 and pkg 20.11.1. I really don't understand. 

`
Error [ERR_REQUIRE_ESM]: require() of ES Module C:\snapshot\Prosec\backend\node_modules\ltx\lib\parsers\ltx.js from C:\snapshot\Prosec\backend\node_modules\node-opcua-xml2json\dist\source\xml2json.js not supported.
ltx.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
`

**Comment by armgong** [comment]
btw, run the orign pkg 5.8.1,there are no error.
but use yao-pkg 5.11.4 or latest version both have similar error, the code itself is same.

**Reply by robertsLando** [reply]
@armgong if you use node16 you will see that it works with yao-pkg as well, the issue is related to changes on nodejs env

**Reply by robertsLando** [reply]
Ref #16

**Reply by armgong** [reply]
thank you very much.
tried different version, finally use yao-pkg 5.9.2. its use node18.18.2,there are no problem on windows.
but it will failed when nodejs >=18.19

**Reply by robertsLando** [reply]
That's expected as explained on #16
